armi/reactor/reactors.py

```diff
@@ -223,6 +220,98 @@ def setOptionsFromCs(self, cs):
+            the value is interpreted as the group-wise delayed neutron
+            fractions.
+            """
+            if value is None:
+                return None
+
+            if isinstance(value, list):
+                return [float(val) for val in value]
+            elif isinstance(value, float):
+                return value
+            else:
+                raise ValueError(
+                    "`beta` must be set as a float or a list of floats. "
+                    f"Attempted to assign {value}."
+                )
+
+        def _decayConstantsValidator(value):
+            """Validate the input for precursor decay constants."""
+            if value is None:
+                return None
+
+            if isinstance(value, list):
+                return [float(val) for val in value]
+            else:
+                raise ValueError(
+                    "`decayConstants` must be set as a list of floats. "
+                    f"Attempted to assign {value}."
+                )
+
+        runLog.info(
+            "Checking for user-supplied `beta` and `decayConstants` settings and attempting to "
+            f"apply them to state of {self}."
+        )
+        beta = _betaValidator(cs["beta"])
+        decayConstants = _decayConstantsValidator(cs["decayConstants"])
+
+        # If beta is interpreted as a single float value, then assign it to
+        # the total delayed neutron fraction parameter. Otherwise, setup the
+        # group-wise delayed neutron fractions and precursor decay constants.
+        reportTableData = []
+        if isinstance(beta, float):
+            self.p.beta = beta
+            reportTableData.append(("Total Delayed Neutron Fraction", self.p.beta))
+
+        elif isinstance(beta, list) and isinstance(decayConstants, list):
+            if len(beta) != len(decayConstants):
+                raise ValueError(
+                    f"The values for `beta` ({beta}) and `decayConstants` "
+                    f"({decayConstants}) are not consistent lengths."
+                )
+
+            self.p.beta = sum(beta)
+            self.p.betaComponents = numpy.array(beta)
+            self.p.betaDecayConstants = numpy.array(decayConstants)
+
+            reportTableData.append(("Total Delayed Neutron Fraction", self.p.beta))
+            reportTableData.append(
+                ("Group-wise Delayed Neutron Fractions", self.p.betaComponents)
+            )
+            reportTableData.append(
+                ("Group-wise Precursor Decay Constants", self.p.betaDecayConstants)
+            )
+
+        # Report to the user the values were not applied.
+        if not reportTableData and (beta is not None or decayConstants is not None):
+            runLog.warning(
+                f"Delayed neutron fraction(s) - {beta} and decay constants - {decayConstants} have not been applied."
+            )
+            return
+
+        runLog.info(
```

Comment:
I think an `else` would be more clear here than an early return

Reply:
Sounds good.